### PR TITLE
tests: context: add a testcase to verify the key of ISR.

### PR DESCRIPTION
**Pull request by Zhaoningx**
Add a testcase to verify the API k_cpu_atomic_idle() can
keep the bit of ISR state unchanged before and after
the call.

Signed-off-by: NingX Zhao <ningx.zhao@intel.com>

### CODE REVIEW

File: tests/kernel/context/src/main.c [hunk 1]
```diff
@@ -1161,6 +1161,27 @@ void test_kernel_thread(void)
 
 }
 
+void test_kernel_cpu_idle_irq_key(void)
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:1164]: Please add the brief and details, thanks!

File: tests/kernel/context/src/main.c [hunk 2]
```diff
@@ -1161,6 +1161,27 @@ void test_kernel_thread(void)
...
+{
+	unsigned int key = 0, mask = 0xF00;
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:1166]: I think this mask should be 0x200, the IF flag.

File: tests/kernel/context/src/main.c [hunk 3]
```diff
@@ -1161,6 +1161,27 @@ void test_kernel_thread(void)
...
+
+	key = irq_lock();
+	zassert_true((key & mask) == 0x200, "The key value is not expected");
+	k_cpu_atomic_idle(key);
+	key = irq_lock();
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:1171]: This code logic looks weird for me. I would suggest we verify that the status of irq lock can be stored and recovered after k_cpu_atomic_idle() was called. Let's have a discuss for this.

File: tests/kernel/context/src/main.c [hunk 4]
```diff
@@ -1161,6 +1161,27 @@ void test_kernel_thread(void)
 
 }
 
+void test_kernel_cpu_idle_irq_key(void)
+{
+	unsigned int key = 0, mask = 0xF00;
+
+	key = irq_lock();
+	zassert_true((key & mask) == 0x200, "The key value is not expected");
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:1169]: I think this should be:  zassert_true((key & mask), "The key value is not expected");

File: tests/kernel/context/src/main.c [hunk 5]
```diff
@@ -1161,6 +1161,27 @@ void test_kernel_thread(void)
...
+	k_cpu_atomic_idle(key);
+	key = irq_lock();
+	zassert_true((key & mask) == 0x200, "The key value is not expected");
+
+	/* locked irq first */
+	irq_lock();
+
+	key = irq_lock();
+	zassert_true((key & mask) == 0x0, "The key value is not expected");
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:1178]: I think this should be:  zassert_false((key & mask), "The key value is not expected");

**Review by enjiamai — CHANGES_REQUESTED** [review]

**Comment by Zhaoningx** [comment]
Hi all, I integrated the code to another PR https://github.com/zephyrproject-rtos/zephyr/pull/39977, so  close this one. Thanks.

**Pull request closed without merging**